PR: https://github.com/odoo/odoo/pull/104223

From: 77f9ff50db3cdb88397d0b1cc7042c772d0d417b
From: Florian Charlier
Date: 2023-06-30 21:37:50

Breaking data model changes scores: del:1 + add:0, change matches:
-    payment_provider_onboarding_state = fields.Selection(

Total Changes: 29

[REF] *: use onboarding module

* = account{_payment}, base, onboarding, payment{_stripe},
sale{_management}, web, website_sale

Use the dedicated onboarding module introduced in 16.0 instead of
the res.company model to store onboarding progress.

It allows
 * onboarding steps to be reused across panels
 * to support steps that should be completed per-database or per-company
 * to clean the res.company model from many fields and methods,
 * to remove many views, controllers, actions

Module-specific notes:
* account: We also clean the remaining two steps that are not
part of an accounting panel but make the most sense to be kept here.
* account_payment: Following 8e4e8eb8, the payment provider step is
added to the invoicing onboarding panel. We apply this change here too.
Also impacts the website_sale_dashboard panel (see related ENT PR).
(The "sale tax" one is currently used for to the website sale dashboard).
* payment: Note that the step was already not part of an onboarding
panel within this module.
* website_sale: We clean
  * a field not used (The website_sale dashboard onboarding panel used
  the payment_provider_onboarding_state field).
  * a method that was only called from website_sale_dashboard, so it is
  moved there. See related ENT PR.

Includes a few tests.

Moving views/templates/styling, as well as cleaning residual onboarding-related fields and methods in base, including populate.

This also includes restoring the "onboarding_complete" overlay panel
animating it to disappear after a few seconds so that it doesn't hide
text and block buttons to re-open steps.

Task-3025136

Part-of: odoo/odoo#104223

================================= pseudo patch: =================================

--- a/addons/payment/models/__init__.py
+++ b/addons/payment/models/__init__.py
@@ -5,5 +5,6 @@ from . import payment_method
 from . import payment_provider
 from . import payment_token
 from . import payment_transaction
+from . import onboarding_onboarding_step
 from . import res_company
 from . import res_partner

--- a/None
+++ b/addons/payment/models/onboarding_onboarding_step.py
@@ -0,0 +1,12 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class OnboardingStep(models.Model):
+    _inherit = 'onboarding.onboarding.step'
+
+    @api.model
+    def action_validate_step_payment_provider(self):
+        """Override to validate other steps as well."""
+        return self.action_validate_step("payment.onboarding_onboarding_step_payment_provider")

--- a/addons/payment/models/res_company.py
+++ b/addons/payment/models/res_company.py
@@ -6,10 +6,6 @@ from odoo import api, fields, models
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
-    payment_provider_onboarding_state = fields.Selection(
-        string="State of the onboarding payment provider step",
-        selection=[('not_done', "Not done"), ('just_done', "Just done"), ('done', "Done")],
-        default='not_done')
     payment_onboarding_payment_method = fields.Selection(
         string="Selected onboarding payment method",
         selection=[
@@ -56,15 +52,3 @@ class ResCompany(models.Model):
         modules_sudo = self.env['ir.module.module'].sudo().search([('name', 'in', module_names)])
         STATES = ['installed', 'to install', 'to upgrade']
         modules_sudo.filtered(lambda m: m.state not in STATES).button_immediate_install()
-
-    def _mark_payment_onboarding_step_as_done(self):
-        """ Mark the payment onboarding step as done.
-
-        :return: None
-        """
-        self.set_onboarding_step_done('payment_provider_onboarding_state')
-
-    def get_account_invoice_onboarding_steps_states_names(self):
-        """ Override of account. """
-        steps = super().get_account_invoice_onboarding_steps_states_names()
-        return steps + ['payment_provider_onboarding_state']
